src/cppnv/EnvKey.h

```diff
@@ -0,0 +1,41 @@
+#ifndef SRC_CPPNV_ENVKEY_H_
+#define SRC_CPPNV_ENVKEY_H_
+#include <string>
+
+namespace cppnv {
+class EnvKey {
+ public:
+  std::string* key;  // this will be the temp buffer they all share
```

Comment:
Use std::string_view for these instead?

Reply:
oh wait, I see `std::string* own_buffer` is being deleted. Why not simply have these as `std::string` and not pointers and rely on the normal cleanup?

Reply:
I'm doing some tricks with buffer re-use here during the interpolation stage and skipping them if there are no interpolations.